### PR TITLE
Fixed for issue#38208

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#38208](https://github.com/Azure/azure-sdk-for-java/issues/38208)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by v-hongli1** [comment]
The new test cases are set to `@DoNotRecord(skipInPlayback = true)`, the reason for ffollowing as：
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3393330&view=results

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java [hunk 1]
```diff
@@ -742,6 +785,52 @@ interface WithManagedServiceIdentity {
             Update withSystemAssignedManagedServiceIdentity();
         }
 
+        /**
+         * The stage of the User Assigned Managed Service Identity enabled storage account allowing to set
+         * access role for the identity.
+         */
+        interface WithUserAssignedManagedServiceIdentityAccessOrUpdate extends Update {
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java:792]: Can this be merged with `WithUserAssignedManagedServiceIdentity`?

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 2]
```diff
@@ -266,4 +269,136 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+            .withUserAssignedManagedServiceIdentity()
+            .withExistingUserAssignedManagedServiceIdentity(defaultIdentity)
+            .create();
+        Assertions.assertEquals(IdentityType.USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
+
+        storageAccount.update().withoutUserAssignedManagedServiceIdentity(defaultIdentity.id()).apply();
+        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
+    }
 }
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:404]: Does storage account support having both system assigned identity and user assigned identity?

Could you add one more test of
1. create storage account with system assigned identity and 2 user assigned identity
2. remove 1 user assigned identity (verify the type and count of remaining user assigned identity)
3. remove the system assigned identity (verify the type)

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:633]: The test cases you want has been added to the new version.

File: sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md [hunk 3]
```diff
@@ -4,11 +4,7 @@
 
 ### Features Added
 
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+- Supported user-assigned service identity for `StorageAccount` in create and update.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/CHANGELOG.md:7]: Please add a block about added dependency, e.g.

```
### Other Changes

- Added dependency of `azure-resourcemanager-msi` and `azure-resourcemanager-authorization`.
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java [hunk 4]
```diff
@@ -872,4 +933,5 @@ public PrivateEndpointConnectionProvisioningState provisioningState() {
             return provisioningState;
         }
     }
+
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java:936]: revert this kind of unnecessary change.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:1]: Could you let us know what's the major difference of the class, compared to other e.g. `VirtualMachineMsiHandler`? We'd like to focus on the code that's different.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java [hunk 5]
```diff
@@ -459,6 +476,7 @@ public StorageAccountImpl update() {
         updateParameters = new StorageAccountUpdateParameters();
         this.networkRulesHelper = new StorageNetworkRulesHelper(this.updateParameters, this.innerModel());
         this.encryptionHelper = new StorageEncryptionHelper(this.updateParameters, this.innerModel());
+        this.storageAccountMsiHandler = new StorageAccountMsiHandler(this.authorizationManager, this, createParameters, this.updateParameters, isInCreateMode());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java:479]: Why do we need a new MsiHandler instance in `update()`?

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:1]: Agree. Seems to me the constructor signature has changed.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:1]: 1. The `StorageAccountMsiHandler` is consistent with the `VirtualMachineMsiHandler` business process.
2. The creation and update data of `Identity` were stored in the respective `StorageAccountCreateParameters` and `StorageAccountUpdateParameters` class.
3. The `StorageAccountInner` works with the existing `Identity` business data provider.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java (hunk 4 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java (hunk 5 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: Here you need to reference the `Storage Account` data in real time.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: Should you instead make `createParameters`, `updateParameters`, `isCreateMode()` in StorageAccountImpl package private, and do `StorageAccountMsiHandler(authorizationManager, this)`. Where in `StorageAccountMsiHandler` you call there variable/method in real time?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: The following adjustments have been made in the `StorageAccountMsiHandler` class:
1. The `isInCreateMode` fixed to get value via private method.
2. The construction method is simplified to `StorageAccountMsiHandler(final AuthorizationManager authorizationManager, StorageAccountImpl storageAccount)`
3. `updateParameters` and `createParameters` are changed to be set in the `withXXXXXXXX` methods, e.g. `withUpdateParameters`.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java [hunk 6]
```diff
@@ -64,12 +67,16 @@ class StorageAccountImpl
...
+    private final AuthorizationManager authorizationManager;
 
-    StorageAccountImpl(String name, StorageAccountInner innerModel, final StorageManager storageManager) {
+    StorageAccountImpl(String name, StorageAccountInner innerModel, final StorageManager storageManager, final AuthorizationManager authorizationManager) {
         super(name, innerModel, storageManager);
+        this.authorizationManager = authorizationManager;
         this.createParameters = new StorageAccountCreateParameters();
         this.networkRulesHelper = new StorageNetworkRulesHelper(this.createParameters);
         this.encryptionHelper = new StorageEncryptionHelper(this.createParameters);
+        this.storageAccountMsiHandler = new StorageAccountMsiHandler(authorizationManager, this).withCreateParameters(createParameters);
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java:79]: Could you try making createParameters and updateParameters package-private in `StorageAccountImpl`? 
This way, you can refer them by `storageAccount.createParameters` and `storageAccount.updateParameters` in `MsiHandler`, and no need for `withCreateParameters` and `withUpdateParameters`.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 7]
```diff
@@ -256,40 +252,28 @@ private void initStorageAccountIdentity(IdentityType identityType) {
...
             } else {
-                updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+                storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
             }
         }
     }
 
     private Boolean isInCreateMode() {
         return Objects.isNull(storageAccount.innerModel().id());
     }
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:278]: `storageAccount.isInCreateMode()`?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: The method `isInCreateMode()` removed from `StorageAccountMsiHandler`.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java [hunk 8]
```diff
@@ -456,7 +473,7 @@ private void clearWrapperProperties() {
     @Override
     public StorageAccountImpl update() {
         createParameters = null;
-        updateParameters = new StorageAccountUpdateParameters();
+        updateParameters = new StorageAccountUpdateParameters().withIdentity(this.innerModel().identity());
```
[the remark this answers is not in the document]
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: Through temporary test case testing, it was found that the change `updateParameters = new StorageAccountUpdateParameters().withIdentity(this.innerModel().identity());` is indeed inappropriate and has been fixed in the new version.

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java]: DONOT DO FORCE PUSH on PR already in review. I cannot see what's the change on this...

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+            } else {
+                storageAccount.createParameters.withIdentity(new Identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED));
+            }
+        } else {
+            if (storageAccount.innerModel().identity() == null
+                || storageAccount.innerModel().identity().type() == null
+                || storageAccount.innerModel().identity().type().equals(IdentityType.NONE)
+                || storageAccount.innerModel().identity().type().equals(identityType)) {
+                Identity identity = Objects.isNull(storageAccount.innerModel().identity()) ? new Identity().withType(identityType) : storageAccount.innerModel().identity().withType(identityType);
+                storageAccount.updateParameters.withIdentity(identity);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:266]: 1. This part is correct to check innerModel (as current Identity configure).

2. However, it then update `updateParameters`.

3. Next time it get called, it would again check innerModel (that does not contain the "update" in (2)).

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: Should we do here similar to `withoutLocalManagedServiceIdentity()`?

Have this above the block
```java
        if (Objects.isNull(storageAccount.updateParameters.identity())) {
            storageAccount.updateParameters.withIdentity(storageAccount.innerModel().identity());
        }
```

And then just check/update `updateParameters`.

@XiaofeiCao Thought?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: Yeah, I think we should. 
Though current code's behavior is the same, it only has implicit indication that `innerModel()` and `updateParameters` share the same `Identity` instance, which makes the code hard to reason about.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+            if (currentIdentity == null
+                || currentIdentity.type() == null
+                || currentIdentity.type().equals(IdentityType.NONE)
+                || currentIdentity.type().equals(identityType)) {
+                currentIdentity = Objects.isNull(currentIdentity) ? new Identity().withType(identityType) : currentIdentity.withType(identityType);
+            } else {
+                currentIdentity.withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+            }
+            this.storageAccount.innerModel().withIdentity(currentIdentity);
+            this.storageAccount.updateParameters.withIdentity(currentIdentity);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:279]: Does this work?

```java
            if (storageAccount.updateParameters.identity() == null) {
                storageAccount.updateParameters.withIdentity(storageAccount.innerModel().identity());
            }
            if (storageAccount.updateParameters.identity() == null) {
                storageAccount.updateParameters.withIdentity(new Identity());
            }

            if (storageAccount.updateParameters.identity().type() == null
                || storageAccount.updateParameters.identity().type().equals(IdentityType.NONE)
                || storageAccount.updateParameters.identity().type().equals(identityType)) {
                storageAccount.updateParameters.identity().withType(identityType);
            } else {
                storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
            }
```

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: The code related to `storageAccount.innerModel().identity()` has been deleted in the new version.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+                } else {
+                    this.storageAccount.updateParameters.identity().withType(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED);
+                }
+            }
+        } else {
+            if (Objects.isNull(this.storageAccount.updateParameters.identity())) {
+                this.storageAccount.updateParameters.withIdentity(this.storageAccount.innerModel().identity());
+            }
+            if (IdentityType.SYSTEM_ASSIGNED.equals(identityType)) {
+                if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.innerModel().identity().type())) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:282]: This line seems still need to check `storageAccount.updateParameters.identity().type()`?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: `if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.innerModel().identity().type()))` corrected to `if (IdentityType.SYSTEM_ASSIGNED.equals(this.storageAccount.updateParameters.identity().type()))`

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+    /** Clear StorageAccountMsiHandler post-run specific internal state. */
+    void clear() {
+        this.userAssignedIdentities = new HashMap<>();
+    }
+
+    /**
+     * Method that handle the case where user request indicates all it want to do is remove all identities associated
+     * with the storage account.
+     *
+     * @return true if user indented to remove all the identities.
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:180]:
```suggestion
     * @return true if user intends to remove all the identities.
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+        }
+    }
+
+    /** Clear StorageAccountMsiHandler post-run specific internal state. */
+    void clear() {
+        this.userAssignedIdentities = new HashMap<>();
+    }
+
+    /**
+     * Method that handle the case where user request indicates all it want to do is remove all identities associated
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java:177]:
```suggestion
     * Method that handles the case where user request indicates all he/she wants to do is to remove all associated identities 
```

File: sdk/resourcemanager/azure-resourcemanager-msi/pom.xml [hunk 14]
```diff
@@ -76,12 +76,6 @@
       <version>1.13.11</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
       <scope>test</scope>
     </dependency>
-    <dependency>
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-msi/pom.xml:79]: why we remove this? no dependency now?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-msi/pom.xml:79]: Yes, not only is there no dependency, it also forms an infinite loop of dependencies between msi and storage.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java (hunk 13 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountMsiHandler.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 15]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
+
+        storageAccount.update()
+            .withNewUserAssignedManagedServiceIdentity(identityCreatable).apply();
+
+        Assertions.assertEquals(IdentityType.USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityPrincipalId());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:388]:
```suggestion
        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 16]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+            .define(saName)
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingUserAssignedManagedServiceIdentity(defaultIdentity)
+            .create();
+
+        Assertions.assertEquals(IdentityType.USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityPrincipalId());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:413]:
```suggestion
        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 17]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+        Assertions.assertNull(storageAccount.innerModel().identity());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityPrincipalId());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
+
+        storageAccount.update()
+            .withExistingUserAssignedManagedServiceIdentity(defaultIdentity).apply();
+
+        Assertions.assertEquals(IdentityType.USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:443]:
```suggestion
        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 18]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+            .storageAccounts()
+            .define(saName)
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingUserAssignedManagedServiceIdentity(defaultIdentity)
+            .create();
+        Assertions.assertEquals(IdentityType.USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityPrincipalId());
+        Assertions.assertNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:467]:
```suggestion
        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 19]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
+            .define(saName)
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withSystemAssignedManagedServiceIdentity()
+            .withExistingUserAssignedManagedServiceIdentity(defaultIdentity)
+            .create();
+        Assertions.assertEquals(IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, storageAccount.innerModel().identity().type());
+        Assertions.assertNotNull(storageAccount.systemAssignedManagedServiceIdentityPrincipalId());
+        Assertions.assertNotNull(storageAccount.systemAssignedManagedServiceIdentityTenantId());
+        Assertions.assertNotNull(storageAccount.userAssignedManagedServiceIdentityIds());
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:497]:
```suggestion
        Assertions.assertTrue(storageAccount.userAssignedManagedServiceIdentityIds().isEmpty());
```

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 20]
```diff
@@ -266,4 +269,447 @@ public void canDisableDefaultToOAuthAuthenticationOnStorageAccount() {
...
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:527]: Same as above, please check them all.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java]: checked them all and fixed in the new version.

**Review by XiaofeiCao — APPROVED** [review]
